lib/internal/url.js

```diff
@@ -839,20 +882,26 @@ class URL {
     this.#context.hash_start = hash_start;
     this.#context.scheme_type = scheme_type;
 
-    if (this.#searchParams) {
-      if (this.#context.hasSearch) {
-        setURLSearchParams(this.#searchParams, this.search);
+    // If URLSearchParams exists, and the search has changed, update it.
+    // Otherwise, check if URLSearchParams has been modified.
+    if (updateSearchParams) {
```

Comment:
I recommend having a variable name without a verb in front of it. It hints me towards a function, but this is a variable.

```suggestion
    if (shouldUpdateSearchParams) {
```